src/features/click-to-load.js

```diff
@@ -1699,48 +1707,34 @@ function createYouTubePreview (originalElement, widget) {
     return { youTubePreview, shadowRoot }
 }
 
-// Convention is that each function should be named the same as the sendMessage
-// method we are calling into eg. calling `sendMessage('getClickToLoadState')`
-// will result in a response routed to `updateHandlers.getClickToLoadState()`.
-const messageResponseHandlers = {
-    getClickToLoadState (response) {
-        devMode = response.devMode
-        isYoutubePreviewsEnabled = response.youtubePreviewsEnabled
-
-        // Mark the feature as ready, to allow placeholder replacements to
-        // start.
-        readyToDisplayPlaceholdersResolver()
-    },
-    setYoutubePreviewsEnabled (response) {
-        if (response?.messageType && typeof response?.value === 'boolean') {
-            originalWindowDispatchEvent(
-                createCustomEvent(
-                    response.messageType, { detail: response.value }
-                )
-            )
-        }
-    },
-    getYouTubeVideoDetails (response) {
-        if (response?.status && typeof response.videoURL === 'string') {
-            originalWindowDispatchEvent(
-                createCustomEvent(
-                    'ddg-ctp-youTubeVideoDetails',
-                    { detail: response }
-                )
-            )
+export default class ClickToLoad extends ContentFeature {
+    // Messaging layer between Click to Load and the Platform
```

Comment:
Does the new messaging abstraction not take of this stuff for us? Seems bad that every feature needs to implement this, checking for platform support of messaging etc?

Reply:
We could move this method to `ContentFeature`, it has access to the featureName as `this.name`

Reply:
I guess because of the `SendMessageMessagingConfig` being temporary we might not want to yet? Happy either way, `ContentFeature` is where it should live eventually for sure.

Reply:
I've not looked at ContentFeature, but if you think that's a better home for the logic then it sounds good to me. It seems kind of hacky having it in the Click to Load code.

Reply:
@kzar yep, it's a work around for now since not all platforms are implemented (yet!) - once we've worked around the patterns for CTL extension/android we can back-port to the messaging lib and make this more seamless.

On a call with @franfaccin today we moved the most boiler-plately parts into ContentFeature already :)

https://github.com/duckduckgo/content-scope-scripts/compare/shane/pr-review?expand=1#diff-cb668ba12d63994b1012861678e5aefba65ddcbe151a3e38a67962d4ab1f7861R46-R55